### PR TITLE
Combobox: :bug: implement Readonly + disabled

### LINKED ISSUE #3049: [BUG]: Combobox sin readOnly-prop virker ikke
## Hva er feilen/problemet?

ReadOnly på Combobox-komponenten fungerer ikke. Den fungerer helt likt med readOnly=true som false. 

## Steg for å reprodusere problemet (valgfri)

Har laget et lite eksempel for å vise at det ikke har noe effekt å sette readOnly til true.
image
image

## Eksempel/Tilleggsinformasjon (valgfri)

```
 Hvis mulig, legg gjerne til kode her, evt lenke til kode i repo
```

### PR DESCRIPTION
**Pull request by JulianNymark**
### Description

Both `readonly` and `disabled` were not working for Combobox. This PR aims to implement both of those in one fell swoop (mostly since they are similar-ish features).

Closes: https://github.com/navikt/team-aksel/issues/645
Closes: https://github.com/navikt/aksel/issues/3049

### Component Checklist 📝

- [x] JSDoc
- [x] Examples
- [x] Documentation
- [x] Storybook
- [x] Style mappings (`@navikt/core/css/config/_mappings.js`)
- [x] Component tokens (`@navikt/core/css/tokens.json`)
- [x] CSS class deprecations (`@navikt/aksel-stylelint/src/deprecations.ts`)
- [x] Exports (`@navikt/core/react/src/index.ts` and `@navikt/core/react/package.json`)
- [x] New component? CSS import (`@navikt/core/css/index.css`)
- [x] Breaking change? Update migration guide. Consider codemod.
- [x] Changeset (Format: `<Component>: <gitmoji?> <Text>.` E.g. "Button: :sparkles: Add feature xyz.")

### CODE REVIEW

**Comment by JulianNymark** [comment]
image

File: .changeset/ten-onions-compare.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+"@navikt/ds-react": patch
+"@navikt/ds-css": patch
+---
+
+:bug: [combobox] implement missing readonly and disabled
```
Comment (HalvorHaugan) [inline .changeset/ten-onions-compare.md]: You didn't follow the new template ;) 
(You may suggest changes to the template if you disagree with it 😄)

File: @navikt/core/react/src/form/combobox/combobox.stories.tsx [hunk 2]
```diff
@@ -493,6 +493,30 @@ export const Disabled: StoryFn = () => {
   );
 };
 
+export const Readonly: StoryFn = () => {
```
Comment (HalvorHaugan) [inline @navikt/core/react/src/form/combobox/combobox.stories.tsx:513]: This story should be added to the Chromatic story.

File: @navikt/core/css/form/combobox.css [hunk 3]
```diff
@@ -20,6 +20,32 @@
   --__ac-combobox-input-height: 1.5rem;
 }
 
+.navds-combobox--disabled {
+  opacity: 0.3;
+}
+
+.navds-combobox--disabled .navds-text-field__input {
+  border: 1px solid var(--a-border-default);
```
Comment (HalvorHaugan) [inline @navikt/core/css/form/combobox.css]: Doesn't it already have this border?

File: @navikt/core/css/form/combobox.css [hunk 4]
```diff
@@ -20,6 +20,32 @@
...
+  cursor: not-allowed;
+}
+
+.navds-combobox--readonly {
+  pointer-events: none;
+}
+
+.navds-combobox--readonly > :where(.navds-form-field__label) {
+  display: inline-flex;
+}
```
Comment (HalvorHaugan) [inline @navikt/core/css/form/combobox.css]: Consider adding the class "navds-form-field--readonly" alongside "navds-combobox--readonly" in ComboboxWrapper, which does the same thing. Then you don't need this CSS.

File: @navikt/core/react/src/form/combobox/Input/Input.tsx [hunk 5]
```diff
@@ -254,7 +259,7 @@ const Input = forwardRef<HTMLInputElement, InputProps>(
           "navds-combobox__input",
           "navds-body-short",
           `navds-body-short--${size}`,
-          { "navds-combobox__input--hide-caret": hideCaret },
+          { "navds-combobox__input--hide-caret": hideCaret || readOnly },
```
Comment (HalvorHaugan) [inline @navikt/core/react/src/form/combobox/Input/Input.tsx]: Is this needed?

File: @navikt/core/react/src/form/combobox/combobox.stories.tsx
Comment (HalvorHaugan) [inline @navikt/core/react/src/form/combobox/combobox.stories.tsx:1]: Can you add `readOnly` to `Default.argTypes`?

Comment (HalvorHaugan) [inline @navikt/core/react/src/form/combobox/combobox.stories.tsx:1]: Can you modify the Disabled story to be more like the Readonly story? (Multi+Single select + values)

File: @navikt/core/react/src/form/combobox/SelectedOptions/SelectedOptions.tsx [hunk 6]
```diff
@@ -28,7 +28,13 @@ const Option = ({ option }: { option: ComboboxOption }) => {
     );
   }
 
-  return <Chips.Removable onClick={onClick}>{option.label}</Chips.Removable>;
+  return readOnly ? (
+    <Chips.Toggle variant="neutral" checkmark={false} as="div">
+      {option.label}
+    </Chips.Toggle>
+  ) : (
+    <Chips.Removable onClick={onClick}>{option.label}</Chips.Removable>
```
Comment (HalvorHaugan) [inline @navikt/core/react/src/form/combobox/SelectedOptions/SelectedOptions.tsx:36]: The chips are clickable when combobox is disabled

File: @navikt/core/react/src/form/combobox/Input/Input.tsx (hunk 5 above)
Comment (JulianNymark) [reply @navikt/core/react/src/form/combobox/Input/Input.tsx]: This was to stop the blinking caret (hiding it) permanently when set to readonly

File: @navikt/core/react/src/form/combobox/Input/Input.tsx
Comment (HalvorHaugan) [inline @navikt/core/react/src/form/combobox/Input/Input.tsx:1]: Consider adding `aria-readonly` to the input field (Looks like `role="combobox"` overrides til semantics of the `readOnly` prop)

File: @navikt/core/css/form/combobox.css (hunk 3 above)
Comment (JulianNymark) [reply @navikt/core/css/form/combobox.css]: This was to override the color on hover, but I think there's a component token for this.

Comment (JulianNymark) [reply @navikt/core/css/form/combobox.css]: actually, nvm, this was completely uneccessary after swapping from using Chip.Removable to Chip.Toggle 👌 Burninating it 🔥

File: @navikt/core/react/src/form/combobox/SelectedOptions/SelectedOptions.tsx (hunk 6 above)
Comment (JulianNymark) [reply @navikt/core/react/src/form/combobox/SelectedOptions/SelectedOptions.tsx:36]: I couldn't actually click them in the example story (probably due to `pointer-events: none` earlier in the DOM tree) 🤔... I'll still set them to be the same as when `readOnly`, despite the fact like Sjur said, nobody should really be using disabled in the first place, and _especially_ with predefined values (that you then can't change) 😅 .

Comment (HalvorHaugan) [reply @navikt/core/react/src/form/combobox/SelectedOptions/SelectedOptions.tsx:36]: But you only have `pointer-events: none` on readOnly?

The Disabled story does not implement `onToggleSelected` which is why nothing happens when you click a chip. You can reproduce it on the default story. You get the hover effect too.

File: @navikt/core/react/src/form/combobox/Input/Input.tsx
Comment (JulianNymark) [reply @navikt/core/react/src/form/combobox/Input/Input.tsx:1]: I've tried setting the `readOnly` native html attribute (which is preferred to `aria-readonly`, but even with and without the role="combobox" attribute, it seems to not be presented to screenreaders as a read-only input? 🤔 

I would like something like this:

```
<input type="text" role="combobox" readOnly />
```

Need to do a bit more experimentation to find a combination of role, type and readonly / aria-readonly that works.

Comment (JulianNymark) [reply @navikt/core/react/src/form/combobox/Input/Input.tsx:1]: So maybe support for readonly is just a bit lackluster currently? 😞, at least we now set it for future support, and readonly is read up from the title on the readonlyIcon.

File: @navikt/core/react/src/form/combobox/Input/Input.tsx (hunk 5 above)
Comment (JulianNymark) [reply @navikt/core/react/src/form/combobox/Input/Input.tsx]: this isn't needed now, removing 🔥

**Review by it-vegard — APPROVED** [review]
LGTM 🚀

**Pull request merged**